unionml/remote.py

```diff
@@ -40,10 +41,23 @@ def create_project(remote: FlyteRemote, project: typing.Optional[str]):
         remote.client.register_project(Project(id=project, name=project, description=project))
 
 
-def get_app_version() -> str:
+def get_app_version(ignore_diff: bool = True) -> str:
```

Comment:
After some reflection, I set the default value to `True` so that the API behaviour is the same (eg. all `get_app_version` call-sites need to opt-in to this check). Not sure what your preference is. The `unionml` cli is the only place that inverts this now, `--ignore-diff` is used to opt out of the check.

Reply:
my inclination is the stricter the better... `ignore_diff = False` by default will blow up a deployment attempt unless the use specifically wants to allow uncommitted changes.

Reply:
I can do that, but #56 is only about failing the deployment. `get_app_version` is used in a bunch of other places, and in some of them it doesn't make sense to fail over a dirty index (eg: commands like `list-model-versions`). It's almost always the `app_version = app_version or get_app_version()` constructs that I think should opt out of this check.

Besides, in the current state of this PR, `get_app_version` gives an action hint on failure talking about `--allow-uncommitted`, which isn't a flag that is present everywhere that `get_app_version` is called. I'll update to `allow_uncommitted = False`, but to keep the rest of the functionality the same, I'll opt those other callsites out for now